libs/core/lib/payment-service.ts

```diff
@@ -0,0 +1,4 @@
+export interface PaymentService {
+  cancel(id: string): Promise<void>;
+  generateUrl(orderId: string, unitAmount: number): Promise<{ id: string, raw: unknown, url: string }>;
```

Comment:
Should we rename it to generatePayment? Because it returns an object not just an url. 

Also let's declare an interface / type above it instead of declaring it inline.